Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Editor/CalibrationTestWindow.cs

```diff
@@ -0,0 +1,521 @@
+                        }
+
+                        if (recordingForRecording.Poses.Count > 0)
+                        {
+                            string rootFolder, targetFolder;
+                            GetRecordingDirectories(out rootFolder, out targetFolder);
+                            string indexPath = Path.Combine(targetFolder, indexFileName);
+
+                            File.WriteAllText(indexPath, JsonUtility.ToJson(recordingForRecording));
+
+                            IndexFilePath = indexPath;
+                        }
+                    }
+                }
+            }
+        }
+
+        private bool IsPlaying
+        {
+            get { return isPlaying; }
+            set
+            {
+                if (isPlaying != value)
+                {
+                    isPlaying = value;
+
+                    if (isPlaying)
+                    {
+                        StartPlayback();
+                    }
+                    else
+                    {
+                        StopPlayback();
+                    }
+                }
+            }
+        }
+
+        private string IndexFilePath
+        {
+            get { return indexFilePath; }
+            set
+            {
+                if (indexFilePath != value)
+                {
+                    indexFilePath = value;
+                    UpdateIndexFileForPlayback();
+                }
+            }
+        }
+
+        private void UpdateIndexFileForPlayback()
+        {
+            isIndexFileParsed = false;
+
+            try
+            {
+                recordingForPlayback = JsonUtility.FromJson<CalibrationRecording>(File.ReadAllText(indexFilePath));
+                isIndexFileParsed = true;
+            }
+            catch
+            {
+            }
+        }
+
+        private string CalibrationFilePath
+        {
+            get { return calibrationFilePath; }
+            set
+            {
+                if (calibrationFilePath != value)
+                {
+                    calibrationFilePath = value;
+                    UpdateCalibrationFileForPlayback();
+                }
+            }
+        }
+
+        private void UpdateCalibrationFileForPlayback()
+        {
+            isCalibrationDataParsed = false;
+            try
+            {
+                if (!string.IsNullOrEmpty(calibrationFilePath))
+                {
+                    byte[] calibrationDataPayload = File.ReadAllBytes(calibrationFilePath);
+
+                    CalculatedCameraCalibration calibration;
+                    if (CalculatedCameraCalibration.TryDeserialize(calibrationDataPayload, out calibration))
+                    {
+                        calibrationDataForPlayback = new CalibrationData(calibration.Intrinsics, calibration.Extrinsics);
+
+                        isCalibrationDataParsed = true;
+                    }
+                }
+            }
+            catch
+            {
+            }
```

Comment:
Is there anything here that would be expected to throw?

nit: it may be worth logging the exception message if we're expecting to catch anything.

Reply:
Added a warning log when this happens, and put a File.Exists check to avoid those expected exceptions when typing in a file name